src/DeterministicFactory_v1.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.23;
+
+// Internal Interfaces
+import {IDeterministicFactory_v1} from
+    "src/interfaces/IDeterministicFactory_v1.sol";
+
+// External Dependencies
+import {Ownable2Step, Ownable} from "@oz/access/Ownable2Step.sol";
+import {Create2} from "@oz/utils/Create2.sol";
+
+/**
+ * @title   Deterministic Deployment Factory
+ *
+ * @notice  This contract is a factory for deploying contracts using the CREATE2 opcode,
+ *             allowing us to deploy some of our contract at the same address of each EVM-
+ *          compatible network.
```

Comment:
It does work for ZK EVM chains, so I won't add anything here. The thing is just that zk EVMs and EVMs don't share the same VM bytecode, and deterministic math operations will only result in the same address if the network uses the same VM bytecode as the other chains you are on. So for all regular EVM chains, our address will be the same (e.g. `0x123...123` and for all ZK EVM chains the address will be the same (e.g. `0.333...333`). 

We'd still use it on ZK EVM chains, as a lot of them share the same VM bytecode, but otherwise, that's just an inherent nature of how CREATE2 works, so I didn't add it here. 

I don't really see the value in adding it here, as it's nothing specific to our factory or approach. But happy to add that if you still think it should be mentioned?